go-controller/pkg/cni/helper_linux.go

```diff
@@ -384,6 +374,79 @@ func (pr *PodRequest) ConfigureInterface(podLister corev1listers.PodLister, kcli
 	return []*current.Interface{hostIface, contIface}, nil
 }
 
+func (pr *PodRequest) UnconfigureInterface(ifInfo *PodInterfaceInfo) error {
+	podDesc := fmt.Sprintf("for pod %s/%s", pr.PodNamespace, pr.PodName)
+	klog.V(5).Infof("Tear down interface (%+v) %s", *pr, podDesc)
+	if pr.CNIConf.DeviceID == "" {
+		if ifInfo.IsDPUHostMode {
+			klog.Warningf("Unexpected configuration %s, Device ID must be present for pod request on smart-nic host",
+				podDesc)
+			return nil
+		}
+		// TODO(gmoodalbail): add similar check for the CX5 ASAP2 VF case
+	}
+
+	// For SRIOV case, we'd need to move the VF from container namespace back to the host namespace
+	if pr.CNIConf.DeviceID != "" {
+		netns, err := ns.GetNS(pr.Netns)
+		if err != nil {
+			return fmt.Errorf("failed to get container namespace %s: %v", podDesc, err)
+		}
+		defer netns.Close()
+
+		hostNS, err := ns.GetCurrentNS()
+		if err != nil {
+			return fmt.Errorf("failed to get host namespace %s: %v", podDesc, err)
+		}
+		defer hostNS.Close()
+
+		err = netns.Do(func(_ ns.NetNS) error {
+			// container side interface deletion
+			link, err := util.GetNetLinkOps().LinkByName(pr.IfName)
+			if err != nil {
+				return fmt.Errorf("failed to get container interface %s %s: %v", pr.IfName, podDesc, err)
+			}
+			err = util.GetNetLinkOps().LinkSetDown(link)
+			if err != nil {
+				return fmt.Errorf("failed to bring down container interface %s %s: %v", pr.IfName, podDesc, err)
+			}
+			// rename VF device back to its original name in the host namespace:
+			if err := util.GetNetLinkOps().LinkSetName(link, ifInfo.VfNetdevName); err != nil {
+				return fmt.Errorf("failed to rename container interface %s to %s %s: %v",
+					pr.IfName, ifInfo.VfNetdevName, podDesc, err)
+			}
+			// move VF device to host netns
+			if err = util.GetNetLinkOps().LinkSetNsFd(link, int(hostNS.Fd())); err != nil {
```

Comment:
Move the err handling to a separate line to be consistent with the above?
 if err != nil {
}

Reply:
done.